src/main/cpp/src/query_operations/variant_operations.cc

```diff
@@ -459,7 +459,7 @@ void remap_allele_specific_annotations(
   const auto merged_non_reference_allele_idx = NON_REF_exists ?
       static_cast<int64_t>(static_cast<int>(num_merged_alleles-1)) : lut_missing_value;
   //index of NON_REF in input sample
-  const auto input_non_reference_allele_idx = NON_REF_exists ?
+  const auto input_non_reference_allele_idx = NON_REF_exists && vid_field_info.remap_missing_with_non_ref() ?
       alleles_LUT.get_input_idx_for_merged(input_call_idx, merged_non_reference_allele_idx) : lut_missing_value;
   //Loop over alleles - only ALT or all alleles (BCF_VL_A or BCF_VL_R)
```

Comment:
Can we generalize this to other fields also? Not just allele specific annotations

Reply:
I started doing this, but had a question when doing some tests for these. Info fields whose length is based on number of alleles (like AD) are sorta easy to reason about with this remapping. But what's the correct behavior for things like PL with length based on number of possible genotypes? Any genotype that includes the missing allele should be missing if the remap is enabled?

Also, it seems like remap_GT_fields also does something like this...but I'm not sure I understand the case where that is required. Won't the merged set of alleles always have the alleles required by each of of the sample's genotypes?

Reply:
> I started doing this, but had a question when doing some tests for these. Info fields whose length is based on number of alleles (like AD) are sorta easy to reason about with this remapping. But what's the correct behavior for things like PL with length based on number of possible genotypes? Any genotype that includes the missing allele should be missing if the remap is enabled?
Yes
> 
> Also, it seems like remap_GT_fields also does something like this...but I'm not sure I understand the case where that is required. Won't the merged set of alleles always have the alleles required by each of of the sample's genotypes?
Right,  GT shouldn't have this issue